pandas/core/series.py

```diff
@@ -2301,6 +2301,19 @@ def diff(self, periods=1):
         dtype: float64
         """
         result = algorithms.diff(com.values_from_object(self), periods)
+        if self.dtype in [
+            "Int8",
+            "Int16",
+            "Int32",
+            "Int64",
+            "UInt8",
+            "UInt16",
+            "UInt32",
+            "UInt64",
+        ]:
+            return self._constructor(
+                result, index=self.index, dtype=self.dtype
```

Comment:
What is the downside of always passing `dtype=self.dtype`?

Reply:
@WillAyd what happens if you do 
```
s = pd.Series([True, False, False]).diff()
```
?

The output will contain a NaN, and if I pass `dtype=self.dtype` in lines 2314-2315 then the resulting series will be of type bool.

It that's what you want, that's fine, but I'll have to change some of the existing tests in pandas/tests/series/methods/test_diff.py